File: app/src/main/java/com/by5388/viewpager/MainActivity.java
package com.by5388.viewpager;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

/**
 * ViewPagerDemo
 *
 * @author by5388  on 2018/12/16.
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        findViewById(R.id.button_fragment).setOnClickListener(v ->
                startActivity(ViewPagerFragmentActivity.newIntent(this))
        );
        findViewById(R.id.button_viewPager).setOnClickListener(v ->
                startActivity(ViewPagerActivity.newIntent(this))
        );
        findViewById(R.id.button_multi_fragment).setOnClickListener(v ->
                startActivity(MultiFragmentActivity.newIntent(this))
        );
    }


}

File: viewpager1/src/main/java/sw/iot/droid/viewpager/MainActivity.kt
package sw.iot.droid.viewpager

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
    }

    private fun initData() {
        val position = intent.getStringExtra(INTENT_KEY)
        textView_show.text = position
    }

    companion object {
        const val INTENT_KEY = "intent_key"
        @JvmStatic
        fun newIntent(context: Context, key: String): Intent {
            val intent = Intent(context, MainActivity::class.java)
            intent.putExtra(INTENT_KEY, key)
            return intent
        }
    }
}
